frontend/src/container/NewWidget/RightContainer/constants.ts

```diff
@@ -11,10 +13,31 @@ export const operatorOptions: DefaultOptionType[] = [
 	{ value: '<=', label: '<=' },
 ];
 
-export const unitOptions = categoryToSupport.map((category) => ({
-	label: category,
-	options: getCategorySelectOptionByName(category),
-}));
+export const getCategorySelectOptionByName = (
+	name?: CategoryNames | string,
+): DefaultOptionType[] =>
+	dataTypeCategories
+		.find((category) => category.name === name)
+		?.formats.map((format) => ({
+			label: format.name,
+			value: format.id,
+		})) || [];
+
+export const unitOptions = (columnUnit: string): DefaultOptionType[] => {
+	const category = getCategoryName(columnUnit);
+	if (isEmpty(category)) {
+		return categoryToSupport.map((category) => ({
+			label: category,
+			options: getCategorySelectOptionByName(category),
+		}));
+	}
+	return categoryToSupport
+		.filter((supportedCategory) => supportedCategory === category)
+		.map((filteredCategory) => ({
+			label: filteredCategory,
+			options: getCategorySelectOptionByName(filteredCategory),
+		}));
+};
```

Comment:
nit: can you move these out of constants.ts?

Reply:
Done